src/main/scala/org/portablescala/sbtplatformdeps/PlatformDepsBuilders.scala

```diff
@@ -0,0 +1,52 @@
+package org.portablescala.sbtplatformdeps
+
+import sbt._
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+final class PlatformDepsGroupID private[sbtplatformdeps] (
+    private val groupID: String) {
+  def %%%(artifactID: String): PlatformDepsGroupArtifactID =
+    macro PlatformDepsGroupID.cross_binary_impl
+}
+
+object PlatformDepsGroupID {
+  /** Internal, used by the macro implementing [[PlatformDepsGroupID.%%%]].
+   *
+   *  Use [[PlatformDepsPlugin.autoImport.platformDepsCrossVersion]] instead.
+   */
+  val platformDepsCrossVersion: SettingKey[CrossVersion] =
+    PlatformDepsPlugin.autoImport.platformDepsCrossVersion
```

Comment:
Is this the right level of abstraction we want to expose to plugins that use this? It seems we are converging towards the additional `_<backend><version>` convention. Have we considered putting all the cross version machinery in here?

Reply:
Actually, we don't know whether we're converging exactly to that convention. Scala Native is not 100% sure how it will deal with architecture- or OS-specific artifacts. This is actually part of the reason for extracting `sbt-platform-deps` off `sbt-crossproject`: sbt-crossproject might break in the future because of this, so we'll make sbt-scalajs depend only on the super stable sbt-platform-deps, and offer sbt-crossproject-scalajs as a separate plugin. The plugin will have the leisure to break less rarely, since it should never be in the middle of the dependency graph of the ecosystem. Only builds will depend on it.

Choosing a `CrossVersion` seems to be the most general solution that `%%%` could possibly depend on, and is therefore basically guaranteed to be stable forever. It also does not seem *too* free-form to me, especially since in sbt 1, `CrossVersion`s are an ADT, so they are restricted themselves, to an extent.

Reply:
Ah, fair enough. In that case `CrossVersion` is good.